server/repo.go

```diff
@@ -238,52 +238,86 @@ func RepairRepo(c *gin.Context) {
 	remote := remote.FromContext(c)
-	repo := session.Repo(c)
+	currentRepo := session.Repo(c)
 	user := session.User(c)
 
+	// verify desination repo query string "to" exists
 	to, exists := c.GetQuery("to")
 	if !exists {
 		err := fmt.Errorf("Missing required to query value")
 		c.AbortWithError(http.StatusInternalServerError, err)
 		return
 	}
 
-	owner, name, errParse := model.ParseRepo(to)
-	if errParse != nil {
-		c.AbortWithError(http.StatusInternalServerError, errParse)
-		return
-	}
-
-	from, err := remote.Repo(user, owner, name)
+	destinationRepo, err := store.GetRepoName(c, to)
 	if err != nil {
 		c.AbortWithError(http.StatusInternalServerError, err)
 		return
 	}
-	if !from.Perm.Admin {
+	perms, permErr := store.PermFind(c, user, destinationRepo)
+	if permErr != nil {
+		c.AbortWithError(http.StatusInternalServerError, permErr)
+		return
+	}
+	// verify user has admin rights on the destination repo and is an admin
+	if !perms.Admin || !user.Admin {
 		c.AbortWithStatus(http.StatusUnauthorized)
 		return
 	}
 
-	repo.Name = from.Name
-	repo.Owner = from.Owner
-	repo.FullName = from.FullName
-	repo.Avatar = from.Avatar
-	repo.Link = from.Link
-	repo.Clone = from.Clone
-	repo.IsPrivate = from.IsPrivate
-	if repo.IsPrivate != from.IsPrivate {
-		repo.ResetVisibility()
+	// swap foreign keys out for builds, secrets, and registry
+	secrets, err := Config.Services.Secrets.SecretList(currentRepo)
+	for _, element := range secrets {
+		element.RepoID = destinationRepo.ID
+		secretsErr := Config.Services.Secrets.SecretUpdate(currentRepo, element)
+		if secretsErr != nil {
+			c.AbortWithError(http.StatusInternalServerError, secretsErr)
+			return
+		}
+	}
+	registries, err := Config.Services.Registries.RegistryList(currentRepo)
+	for _, element := range registries {
+		element.RepoID = destinationRepo.ID
+		registriesErr := Config.Services.Registries.RegistryUpdate(currentRepo, element)
+		if registriesErr != nil {
+			c.AbortWithError(http.StatusInternalServerError, registriesErr)
+			return
+		}
+	}
+	builds, err := store.GetBuildList(c, currentRepo)
+	for _, element := range builds {
+		element.RepoID = destinationRepo.ID
+		buildsErr := store.UpdateBuild(c, element)
+		if buildsErr != nil {
+			c.AbortWithError(http.StatusInternalServerError, buildsErr)
+			return
+		}
+	}
+
+	// Update the destination repo to match current repo settings
+	destinationRepo.Timeout = currentRepo.Timeout
+	destinationRepo.Config = currentRepo.Config
+	if destinationRepo.IsPrivate != currentRepo.IsPrivate {
+		destinationRepo.ResetVisibility()
 	}
+	destinationRepo.Counter = currentRepo.Counter
+	destinationRepo.AllowDeploy = currentRepo.AllowDeploy
+	destinationRepo.AllowPull = currentRepo.AllowPull
+	destinationRepo.AllowPush = currentRepo.AllowPush
+	destinationRepo.AllowTag = currentRepo.AllowTag
+	destinationRepo.IsTrusted = currentRepo.IsTrusted
+	destinationRepo.IsGated = currentRepo.IsGated
+	destinationRepo.IsActive = currentRepo.IsActive
```

Comment:
@bradrydzewski I noticed that isActive doesn't get set from remote.Activate  I imagine I should disable the currentRepo and save that change?